packages/apps/docs/src/components/PageHelpful/useModal.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  Box,
+  Button,
+  Heading,
+  Stack,
+  Text,
+  useModal as useUIModal,
+} from '@kadena/react-ui';
+
+import { EditPage } from '../BottomPageSection/components/EditPage';
+
+import { modalWrapperClass, textAreaClass } from './styles.css';
+
+import { analyticsEvent, EVENT_NAMES } from '@/utils/analytics';
+import { usePathname } from 'next/navigation';
+import React, { useCallback, useRef } from 'react';
+
+interface IUseModalHookResult {
+  renderModalComponent: () => void;
+  comment?: string;
+}
+
+export default function useModal(editLink?: string): IUseModalHookResult {
+  const { renderModal, clearModal } = useUIModal();
+  const inputRef = useRef<HTMLTextAreaElement>(null);
+  const pathname = usePathname();
+
+  const closeModal = (): void => {
+    analyticsEvent(EVENT_NAMES['click:page_helpful'], {
+      pagePath: pathname,
+      isPageHelpful: 'no',
+      comment: inputRef?.current?.value,
+    });
+    clearModal();
+  };
+
+  const renderModalComponent = useCallback((): void => {
+    renderModal(
+      <div className={modalWrapperClass}>
```

Comment:
can u check the design with @isaozler ?
I think the edit button is a bit in a weird position.